myapp/app/models/task.rb

```diff
@@ -1,2 +1,23 @@
+require 'time'
+
 class Task < ApplicationRecord
+  validates :title, :description, presence: { message: 'must not be blank!' }
+  validates :title, length: { in: 5..30, too_short: 'please input more than %{count} characters', too_long: '%{count} characters is the maximum allowed' }
+  validates :description, length: { in: 10..300, too_short: 'please input more than %{count} characters', too_long: '%{count} characters is the maximum allowed' }
+  validates :due, presence: { message: 'must be specified!' }
+
+  validate :due_cannot_be_in_the_past_on_create, on: :create
+  validate :due_cannot_be_changed_to_past_on_update, on: :update
+
+  def due_cannot_be_in_the_past_on_create
+    if due.present? && due < Time.zone.now.beginning_of_minute
+      errors.add(:due, "can't be earlier than now!")
+    end
+  end
+
+  def due_cannot_be_changed_to_past_on_update
+    if due.present? && due_changed? && due < Time.zone.now.beginning_of_minute
```

Comment:
I think we don't have to limit the validate situation with `due_changed?`. Even It is fine to always validate.

Reply:
I'm thinking that, if you always validate due for every edit/update, it will make it impossible to change the content of task (anything but due) after it is created. Otherwise you need to always update the due every time you edit the task.

Reply:
Ahh, I see...
Then how about writing like this?

``` rb
validate :due_cannot_be_in_the_past, if: :due_changed?
```

Reply:
Fixed it! in d842c89

Reply:
8d476a1 please check it